### ISSUE TITLE
Newer / Better Way of Latency Compensation for Hitbox

### ISSUE THREAD

**Issue by Rerd** [konsolas/AAC-Issues#1302]
The movement compensation for the hitbox check isn't perfect right now. Sometimes if it isn't high enough, it will false ban higher latency players, but when you put it too high, it lets some reach cheats bypass. I have a possible alternative to the movement compensation that I stumbled upon on twitter a few months ago. Personally, I think this would do a much better job at compensating for latency.

Anyways, here is the image explaining how this would work: [imgur.com](https://imgur.com/4ckXiOs?) (great drawings)

In short, instead of checking the movement of the player getting hit, it should take the latency of the player, then finding out where they were [player getting hit's ping] ms ago, then calculating the reach from that position.

Opinions would be great, leave a comment below. If we find an issue, maybe we can work out a solution to it.
